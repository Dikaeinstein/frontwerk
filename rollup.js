module.exports = require('./dist/config/rollup.config');
