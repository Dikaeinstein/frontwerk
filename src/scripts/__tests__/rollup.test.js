/* eslint-disable global-require */
import {
  unquoteSerializer,
  winPathSerializer
} from '../../helpers/serializers';

jest.mock('cross-spawn');
jest.mock('concurrently');
jest.mock('jest');
jest.mock('rimraf');

const cases = require('jest-in-case');
const jestSerializerPath = require('jest-serializer-path');

expect.addSnapshotSerializer(unquoteSerializer);
expect.addSnapshotSerializer(winPathSerializer);
expect.addSnapshotSerializer(jestSerializerPath);

const testCases = [
  {
    name: 'calls rollup CLI with default args'
  },
  {
    name: '--no-clean will not clean the dist directory',
    args: ['--no-clean']
  },
  {
    name: 'watches with --watch',
    args: ['--watch']
  },
  {
    name: '--environment will pass the environment',
    args: ['--environment', 'production']
  },
  {
    name: 'does not use built-in config with rollup.config.js file',
    fileExists: filename => filename === 'rollup.config.js'
  },
  {
    name: 'does not use built-in config with --config',
    args: ['--config', './custom-config.js']
  }
];

const testFn = ({ fileExists = () => false, args = [] }) => {
  const { sync: crossSpawnSyncMock } = require('cross-spawn');
  const originalExit = process.exit;
  const originalArgv = process.argv;
  process.exit = jest.fn();

  Object.assign(require('../../utils/fileExists'), { fileExists });
  Object.assign(require('../../utils/resolveBin'), {
    resolveBin: (modName, { executable = modName } = {}) => executable
  });

  process.exit = jest.fn();

  try {
    // tests
    process.argv = ['node', '../rollup', ...args];
    crossSpawnSyncMock.mockClear();

    require('../rollup');

    expect(crossSpawnSyncMock).toHaveBeenCalledTimes(1);
    const [firstCall] = crossSpawnSyncMock.mock.calls;
    const [script, calledArgs] = firstCall;
    expect([script, ...calledArgs].join(' ')).toMatchSnapshot();
  } catch (error) {
    throw error;
  } finally {
    // afterEach
    process.exit = originalExit;
    process.argv = originalArgv;
    jest.resetModules();
  }
};

cases('rollup', testFn, testCases);
