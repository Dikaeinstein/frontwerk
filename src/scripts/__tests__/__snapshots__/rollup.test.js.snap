// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`format --environment will pass the environment 1`] = `"concurrently --names esm,cjs,umd,umd.min \\"cross-env BUILD_ROLLUP=true BUILD_FORMAT=esm BUILD_MINIFY=false NODE_ENV=development  rollup --config ./src/config/rollup.config.js --environment production\\" \\"cross-env BUILD_ROLLUP=true BUILD_FORMAT=cjs BUILD_MINIFY=false NODE_ENV=development  rollup --config ./src/config/rollup.config.js --environment production\\" \\"cross-env BUILD_ROLLUP=true BUILD_FORMAT=umd BUILD_MINIFY=false NODE_ENV=development  rollup --config ./src/config/rollup.config.js --environment production --sourcemap\\" \\"cross-env BUILD_ROLLUP=true BUILD_FORMAT=umd BUILD_MINIFY=true NODE_ENV=production  rollup --config ./src/config/rollup.config.js --environment production --sourcemap\\""`;

exports[`format --no-clean will not clean the dist directory 1`] = `"concurrently --names esm,cjs,umd,umd.min \\"cross-env BUILD_ROLLUP=true BUILD_FORMAT=esm BUILD_MINIFY=false NODE_ENV=development  rollup --config ./src/config/rollup.config.js\\" \\"cross-env BUILD_ROLLUP=true BUILD_FORMAT=cjs BUILD_MINIFY=false NODE_ENV=development  rollup --config ./src/config/rollup.config.js\\" \\"cross-env BUILD_ROLLUP=true BUILD_FORMAT=umd BUILD_MINIFY=false NODE_ENV=development  rollup --config ./src/config/rollup.config.js --sourcemap\\" \\"cross-env BUILD_ROLLUP=true BUILD_FORMAT=umd BUILD_MINIFY=true NODE_ENV=production  rollup --config ./src/config/rollup.config.js --sourcemap\\""`;

exports[`format calls rollup CLI with default args 1`] = `"concurrently --names esm,cjs,umd,umd.min \\"cross-env BUILD_ROLLUP=true BUILD_FORMAT=esm BUILD_MINIFY=false NODE_ENV=development  rollup --config ./src/config/rollup.config.js\\" \\"cross-env BUILD_ROLLUP=true BUILD_FORMAT=cjs BUILD_MINIFY=false NODE_ENV=development  rollup --config ./src/config/rollup.config.js\\" \\"cross-env BUILD_ROLLUP=true BUILD_FORMAT=umd BUILD_MINIFY=false NODE_ENV=development  rollup --config ./src/config/rollup.config.js --sourcemap\\" \\"cross-env BUILD_ROLLUP=true BUILD_FORMAT=umd BUILD_MINIFY=true NODE_ENV=production  rollup --config ./src/config/rollup.config.js --sourcemap\\""`;

exports[`format does not use built-in config with --config 1`] = `"concurrently --names esm,cjs,umd,umd.min \\"cross-env BUILD_ROLLUP=true BUILD_FORMAT=esm BUILD_MINIFY=false NODE_ENV=development  rollup\\" \\"cross-env BUILD_ROLLUP=true BUILD_FORMAT=cjs BUILD_MINIFY=false NODE_ENV=development  rollup\\" \\"cross-env BUILD_ROLLUP=true BUILD_FORMAT=umd BUILD_MINIFY=false NODE_ENV=development  rollup --sourcemap\\" \\"cross-env BUILD_ROLLUP=true BUILD_FORMAT=umd BUILD_MINIFY=true NODE_ENV=production  rollup --sourcemap\\""`;

exports[`format does not use built-in config with rollup.config.js file 1`] = `"concurrently --names esm,cjs,umd,umd.min \\"cross-env BUILD_ROLLUP=true BUILD_FORMAT=esm BUILD_MINIFY=false NODE_ENV=development  rollup --config\\" \\"cross-env BUILD_ROLLUP=true BUILD_FORMAT=cjs BUILD_MINIFY=false NODE_ENV=development  rollup --config\\" \\"cross-env BUILD_ROLLUP=true BUILD_FORMAT=umd BUILD_MINIFY=false NODE_ENV=development  rollup --config --sourcemap\\" \\"cross-env BUILD_ROLLUP=true BUILD_FORMAT=umd BUILD_MINIFY=true NODE_ENV=production  rollup --config --sourcemap\\""`;

exports[`format watches with --watch 1`] = `"concurrently --names esm,cjs,umd,umd.min \\"cross-env BUILD_ROLLUP=true BUILD_FORMAT=esm BUILD_MINIFY=false NODE_ENV=development  rollup --config ./src/config/rollup.config.js --watch\\" \\"cross-env BUILD_ROLLUP=true BUILD_FORMAT=cjs BUILD_MINIFY=false NODE_ENV=development  rollup --config ./src/config/rollup.config.js --watch\\" \\"cross-env BUILD_ROLLUP=true BUILD_FORMAT=umd BUILD_MINIFY=false NODE_ENV=development  rollup --config ./src/config/rollup.config.js --watch --sourcemap\\" \\"cross-env BUILD_ROLLUP=true BUILD_FORMAT=umd BUILD_MINIFY=true NODE_ENV=production  rollup --config ./src/config/rollup.config.js --watch --sourcemap\\""`;
